### PR TITLE
Deprecate classes in package jsonschema

### PR DESCRIPTION
**Pull request by luozhenyu**
The package `com.fasterxml.jackson.databind.jsonschema` has moved to an external module [JSON Schema generator module](https://github.com/FasterXML/jackson-module-jsonSchema) and all classes in this package should be marked with `@Deprecated`.

This PR deprecated classes and methods in package jsonschema. Programmers implementing `interface SchemaAware` or override method `public JsonNode getSchema(SerializerProvider provider, Type typeHint)` will get a deprecation warning.

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java:20]: Not true; was not deprecated in 2.2; this will be against 2.15

File: src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java:31]: Please do not make unrelated changes like this.

File: src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java:41]: Please do not make such mechanical changes; I know it is redundant but these changes obscure actual diff.

File: src/main/java/com/fasterxml/jackson/databind/ser/BeanPropertyWriter.java
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/ser/BeanPropertyWriter.java:876]: No, please leave as-is.

**Comment by cowtowncoder** [comment]
Ok; let's please start with a smaller schema just deprecating `getSchema()` method -- easier to review.

File: src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java
Comment (luozhenyu) [reply src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java:20]: I copied this javadoc from `com.fasterxml.jackson.databind.jsonschema.JsonSchema`. I would have thought that `JsonSerializableSchema` and `SchemaAware` were deprecated since 2.2 along with `JsonSchema` but just forgot to mark with deprecated. So It should be against to 2.15?
https://github.com/FasterXML/jackson-databind/blob/2.15/src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSchema.java#L19

File: src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java
Comment (luozhenyu) [reply src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java]: I got it

File: src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java
Comment (pjfanning) [reply src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java:20]: you can't just pretend this was deprecated back in v2.2 - you have to say 2.15

File: src/main/java/com/fasterxml/jackson/databind/ext/DOMSerializer.java
Comment (pjfanning) [inline src/main/java/com/fasterxml/jackson/databind/ext/DOMSerializer.java:54]: add javadoc with `@deprecated since 2.15` in the javadoc (as well)

File: src/main/java/com/fasterxml/jackson/databind/ser/BeanPropertyWriter.java
Comment (pjfanning) [reply src/main/java/com/fasterxml/jackson/databind/ser/BeanPropertyWriter.java:876]: just revert this change

File: src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java
Comment (cowtowncoder) [reply src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java:20]: Right, @luozhenyu it was not declared as deprecated and we cannot "backdate" decision -- even if that might have been intent. I suspect it was not deprecated just to reduce noise but either way `2.15` is needed here.

File: src/main/java/com/fasterxml/jackson/databind/ser/BeanPropertyWriter.java
Comment (cowtowncoder) [reply src/main/java/com/fasterxml/jackson/databind/ser/BeanPropertyWriter.java]: @luozhenyu I appreciate your avoiding warnings, but is this the only way? (I forget if there was another mechanism).

File: src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java
Comment (luozhenyu) [reply src/main/java/com/fasterxml/jackson/databind/jsonschema/JsonSerializableSchema.java]: Thanks, I have changed it to 2.15

**Review by cowtowncoder — APPROVED** [review]

**Comment by cowtowncoder** [comment]
Ok @luozhenyu this looks good: thank you for contributing the PR!

The last before I merge this, unless I have asked and received it (apologies if so, just let me know), is Contributor License Agreement (CLA) from https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf .
It is usually easiest to print, fill and sign, scan/photo, email to `info` at fasterxml dot com.
Once I have it I can proceed with merging the PR. CLA only needs to be sent once and is good for all future PRs.

Looking forward to merging this!

**Comment by luozhenyu** [comment]
> Ok @luozhenyu this looks good: thank you for contributing the PR!
> 
> The last before I merge this, unless I have asked and received it (apologies if so, just let me know), is Contributor License Agreement (CLA) from https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf . It is usually easiest to print, fill and sign, scan/photo, email to `info` at fasterxml dot com. Once I have it I can proceed with merging the PR. CLA only needs to be sent once and is good for all future PRs.
> 
> Looking forward to merging this!

I have signed and sent by email~

**Review by cowtowncoder — APPROVED** [review]

**Pull request merged**
